Make the values endpoint streaming again

solsson opened the pull request:
This avoids memory usage spikes when retrieving large values responses. Reverts parts of #48 but also reduces the test coverage that #48 introduced. I found no easy way to render a response body from the "entity" and I didn't feel like introducing a REST test framework for this change.

solsson committed:
Restores streaming, to minimize memory spikes for values endpoint

Reverts part of #48 and also raises an overdue review question if
we can retrieve offsets after we've streamed the values.
I think so but that probably depends on threads.

solsson committed:
Current Quarkus 2.x

atamon reviewed (approved):
LGTM, I can only +1 on the the TODO and REVIEW concerns. No we probably cannot guarantee header<->values consistency with this, and yes we should get the test coverage back :D

solsson committed:
Still responds with the toString of the StreamingOutput

solsson committed:
Implmenents streamed response with headers, restores test coverage

It's a gotcha with StreamingOutput that if you also add headers
it's written (by this JAX-RS impl?) as entity.toString

solsson commented:
Replaced by #51 which is a smaller diff and keeps test coverage
